From dca83c9edcba574482ac1a93759b196344dc7a87 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 24 Jun 2015 10:57:07 +1000
Subject: [PATCH libinput 2/4] touchpad: always set touch->palm.time on touch
 begin

We will use this outside of DWT, so set it unconditionally on touch begin.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Hans de Goede <hdegoede@redhat.com>
---
 src/evdev-mt-touchpad.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/evdev-mt-touchpad.c b/src/evdev-mt-touchpad.c
index 9dc5a0e..877e667 100644
--- a/src/evdev-mt-touchpad.c
+++ b/src/evdev-mt-touchpad.c
@@ -211,6 +211,7 @@ tp_begin_touch(struct tp_dispatch *tp, struct tp_touch *t, uint64_t time)
 	t->state = TOUCH_BEGIN;
 	t->millis = time;
 	tp->nfingers_down++;
+	t->palm.time = time;
 	assert(tp->nfingers_down >= 1);
 }
 
@@ -491,7 +492,6 @@ tp_palm_detect_dwt(struct tp_dispatch *tp, struct tp_touch *t, uint64_t time)
 	if (tp->dwt.keyboard_active &&
 	    t->state == TOUCH_BEGIN) {
 		t->palm.state = PALM_TYPING;
-		t->palm.time = time;
 		t->palm.first = t->point;
 		return 1;
 	} else if (!tp->dwt.keyboard_active &&
-- 
2.4.3

